zipline/finance/cancel_policy.py

```diff
@@ -0,0 +1,70 @@
+import abc
+import logbook
+
+from six import with_metaclass, iteritems
+
+from zipline.gens.sim_engine import DAY_END
+from zipline.utils.serialization_utils import (
+    VERSION_LABEL
+)
+
+warning_logger = logbook.Logger('AlgoWarning')
+
+
+def log_warning(amount, sid, filled, policy_name):
+    warning_logger.warn(
+        'Your order for %s shares of %s has been partially filled. %s '
+        'shares were successfully purchased. The remaining %s shares '
+        'are being canceled based on the policy %s.' %
+        (amount, sid, filled, amount - filled, policy_name)
+    )
+
+
+class CancelPolicy(with_metaclass(abc.ABCMeta)):
+
+    def should_cancel(self, dt, event, order):
+        pass
+
+    def __getstate__(self):
+
+        state_dict = \
+            {k: v for k, v in iteritems(self.__dict__)
+                if not k.startswith('_')}
```

Comment:
Why are we skipping private members here?  How would we expect these to be repopulated if we actually had to use this code?